docs/designs/high-level.md

```diff
@@ -1,33 +1,108 @@
 
-# Reconcilers
-1. Initiated by the Triggers components. 
-2. Responsible for executing its logic end to end.
-3. Each Reconciler manage it own status.
+Each custom resource answering to that schema (kind: `OpenSearchCluster`) represents a **single** OpenSearch cluster. 
 
-# Exporters
-1. Export metrics and OS knowledge.
+See [Custom Resource Reference Guide](crd.md) for a full reference to the custom resource.
 
+### Architecture Diagram
 
-Most interactions with the operator will be performed through the OpenSearch operator.
-This can be done in 2 ways:
-1. Updating its CRD. 
-2. Sending API request.
+```mermaid
+  flowchart TD
+      subgraph operator [OpenSearch Operator]
+          direction LR
+          manager[Manager] -->controller
+      end
+      subgraph controller [Controller]
+          direction TB
+          opensearch-reconciler[/OpenSearch Reconciler function/] --Create & execute -->reconcilers
+          subgraph reconcilers [Reconcilers Chain]
+              reconciler1[Reconciler 1] --- reconciler2
+              reconciler2[Reconciler 2] --- reconciler3
+              reconciler3[Reconciler 3] --- reconciler_etc[...]
+          end
+          subgraph components
+              scaler([Scaler component])
+              auto-scaler([Auto-Scaler Component]) 
+          end
+      end
+      
+      controller-- watches -->cr{{OpenSearchCluster Custom Resource}}
+      
+      subgraph kubernetes-objects-created [Created K8s Objects]
+          direction LR
+          opensearch-cluster-statefulsets[OpenSearch Cluster Statefulsets]
+          openseach-config-map[OpenSearch ConfigMap]
+          opensearch-service[OpenSearch Service]
+          ...
+      end    
+
+      subgraph monitoring [Monitoring Tools]
+          prometheus[Prometheus]
+          grafana[Grafana]
+          alert-manager[Alert Manager]
+      end
+      
+      manager --> kubernetes-objects-created
+```
+
+### Provisioning 
+The operator contains a Manager, which manges a Controller for its Custom Resource (`OpenSearchCluster`). It is mainly a listener on queue, which gets filled when ever a change has been made to one of the custom resources. The action is adjusting the deployed cluster to match the desired configuration specified in the custom resource. It can be a simple change to the Security configuration which may only update dynamic settings using OpenSearch API,or it can also be a change to the pod spec, increasing memory requirements from 10gb to 12gb. The latter requires a series of steps to accomplish that.
+
+Once a change has been detected it activates the Controller with the changed custom resource ID, which activates the OpenSearch Reconciler reconcile function. That reconciler is the main entry for the whole operator. Its goal is to modify/create/delete the OpenSearch Cluster and other deployed tools (e.g. Prometheus) reflecting the changes made to the custom resource. 
```

Comment:
I prefer "call" instead of "activate". "activate" for me sounds more like some indirect/async interaction.